code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/Offer.java

```diff
@@ -295,56 +295,100 @@ public void tapped() {
     }
 
     /**
-     * Generates a map containing XDM formatted data for {@code Experience Event -
-     * OptimizeProposition Interactions} field group from this {@code OptimizeProposition} item.
+     * Generates a map containing XDM-formatted data for the {@code Experience Event -
+     * OptimizeProposition Display} field group from this {@code Offer} instance.
      *
-     * <p>The returned XDM data does contain the {@code eventType} for the Experience Event with
-     * value {@code decisioning.propositionDisplay}.
+     * <p>This method constructs a new {@code OptimizeProposition} object using the current offer
+     * and its associated proposition reference. If the {@code scopeDetails} of the referenced
+     * proposition are empty, the {@code activity} and {@code placement} maps are also included;
+     * otherwise, the simpler constructor is used.
      *
-     * <p>Note: The Edge sendEvent API can be used to dispatch this data in an Experience Event
+     * <p>The returned XDM map contains the {@code eventType} {@code
+     * decisioning.propositionDisplay}.
+     *
+     * <p>Note: The Edge sendEvent API can be used to dispatch this data as an Experience Event,
      * along with any additional XDM, free-form data, and override dataset identifier.
      *
-     * @return {@code Map<String, Object>} containing the XDM data for the proposition interaction.
+     * @return {@code Map<String, Object>} containing the XDM data for the proposition display
+     *     interaction.
      * @see XDMUtils#generateInteractionXdm(String, List)
      */
     public Map<String, Object> generateDisplayInteractionXdm() {
         if (propositionReference == null || propositionReference.get() == null) {
             return null;
         }
-        OptimizeProposition proposition =
-                new OptimizeProposition(
-                        propositionReference.get().getId(),
-                        Collections.singletonList(this),
-                        propositionReference.get().getScope(),
-                        propositionReference.get().getScopeDetails());
+
+        OptimizeProposition original = propositionReference.get();
+        OptimizeProposition proposition;
+
+        if (original.getScopeDetails().isEmpty()) {
+            proposition =
+                    new OptimizeProposition(
+                            original.getId(),
+                            Collections.singletonList(this),
+                            original.getScope(),
+                            original.getScopeDetails(),
+                            original.getActivity(),
+                            original.getPlacement());
+        } else {
+            proposition =
+                    new OptimizeProposition(
+                            original.getId(),
+                            Collections.singletonList(this),
+                            original.getScope(),
+                            original.getScopeDetails());
+        }
```

Comment:
Why do we need the if-else block here? Can we just not create OptimizeProposition using all values since `original` would have all values and empty maps for non existent values

Reply:
Ok I think you originally had scopeDetails in the constructor for ODE as well which was confusing. This makes sense